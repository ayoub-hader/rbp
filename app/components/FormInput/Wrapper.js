import styled from 'styled-components';

const Wrapper = styled.div`
  height: 45px;
  position: relative;
  margin-bottom: 15px;
  label {
    top: 0;
    left: 15px;
    position: absolute;
    transform: translate(0, 12px) scale(1);
    font-size: 14px;
    display: none;
    &.affLabel {
      transform: translate(0, 1.5px) scale(0.75);
      font-size: 11px;
      display: block;
      color: ${(props) => props.theme.colors.main};
      font-family: 'source_sans_prosemibold';
    }
  }
  input {
    border-radius: 5px;
    height: 100%;
    padding-left: 15px;
    border: 1px solid ${(props) => props.theme.colors.gray};
    font-size: 16px;
    color: ${(props) => props.theme.colors.main};
    width: 100%;
    &:hover,
    &:focus,
    &.active {
      outline: 0;
      border: 1px solid ${(props) => props.theme.colors.main};
    }
    &::placeholder {
      top: 0;
      left: 15px;
      position: absolute;
      transform: translate(0, 12px) scale(1);
      font-size: 14px;
      font-family: 'source_sans_prosemibold';
      color: ${(props) => props.theme.colors.main};
    }
    &.active {
      &::placeholder {
        transform: translate(0, 1.5px) scale(0.75);
        font-size: 11px;
        color: ${(props) => props.theme.colors.main};
        font-family: 'source_sans_prosemibold';
      }
    }
  }
  .affLabel + input::placeholder {
    text-indent: -9999px;
  }
`;
export default Wrapper;
