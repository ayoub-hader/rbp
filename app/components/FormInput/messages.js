import { defineMessages } from 'react-intl';

export const scope = 'app.components.FormInput';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'This is the FormInput component!',
  },
});
