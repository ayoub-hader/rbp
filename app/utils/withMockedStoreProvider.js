import React from 'react';
import { Provider } from 'react-redux';
import PropTypes from 'prop-types';
import storeMock from 'utils/storeMock';
import { DEFAULT_LOCALE } from 'locales';
import { IntlProvider } from 'react-intl';

const withMockedStoreProvider = (
  Component,
  store = storeMock({}),
  locale = DEFAULT_LOCALE,
) => {
  const TestComponent = props => (
    <Provider store={store}>
      <IntlProvider locale={locale}>
        <Component {...props} />
      </IntlProvider>
    </Provider>
  );

  TestComponent.propTypes = {
    store: PropTypes.object,
    locale: PropTypes.object,
  };
  return TestComponent;
};

export default withMockedStoreProvider;
