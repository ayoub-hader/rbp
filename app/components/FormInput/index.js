/**
 *
 * FormInput
 *
 */

import React, { useState } from 'react';
import PropTypes from 'prop-types';
import Wrapper from './Wrapper';
import { useForm } from 'react-hook-form';
// import styled from 'styled-components';

function FormInput({
  label,
  onChange,
  register,
  value,
  placeholder,
  className,
}) {
  const [isActive, setActive] = useState(false);
  const [affLabel, setAffLabel] = useState(false);
  const handleToggle = () => {
    if ((!value || {}).length > 0) {
      setActive(!isActive);
    }
  };
  const handleOnchange = () => {
    if ((!value || {}).length > -1) {
      setAffLabel(true);
    } else {
      setAffLabel(false);
    }
  };
  const handleBlur = e => {
    if (!e.currentTarget.contains(e.relatedTarget)) {
      if ((!value || {}).length > 0 && isActive) {
        setActive(!isActive);
        setAffLabel(false);
      }
    }
  };

  return (
    <Wrapper>
      <label
        className={(isActive ? 'active' : null, affLabel ? 'affLabel' : null)}
      >
        {label}
      </label>
      <input {...register(label)} />
    </Wrapper>
  );
}

FormInput.propTypes = {
  label: PropTypes.string,
  placeholder: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  onChange: PropTypes.object,
};

export default FormInput;
