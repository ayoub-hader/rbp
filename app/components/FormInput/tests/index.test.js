import React from 'react';
import { render } from '@testing-library/react';
import FormInput from 'components/FormInput/index';
import withMockedStoreProvider from "utils/withMockedStoreProvider";

const FormInputContainer = withMockedStoreProvider(<FormInput />);

describe('<FormInput />', () => {
  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<FormInputContainer />);
    expect(firstChild).toMatchSnapshot();
  })
})
