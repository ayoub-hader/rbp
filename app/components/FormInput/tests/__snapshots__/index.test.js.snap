// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormInput /> should render and match the snapshot 1`] = `
<label
  data-testid="label"
/>
`;
